databricks/koalas/frame.py

```diff
@@ -3222,6 +3222,113 @@ def merge(left, right: 'DataFrame', how: str = 'inner',
         else:
             return DataFrame(selected_columns)
 
+    def join(left, right: 'DataFrame', on: Optional[Union[str, List[str]]] = None,
+             how: str = 'left', lsuffix: str = '', rsuffix: str = '') -> 'DataFrame':
+        """
+        Join columns of another DataFrame.
+
+        Join columns with `right` DataFrame either on index or on a key column. Efficiently join
+        multiple DataFrame objects by index at once by passing a list.
+
+        Parameters
+        ----------
+        right: DataFrame, Series
+        on: str, list of str, or array-like, optional
+            Column or index level name(s) in the caller to join on the index in `right`, otherwise
+            joins index-on-index. If multiple values given, the `right` DataFrame must have a
+            MultiIndex. Can pass an array as the join key if it is not already contained in the
+            calling DataFrame. Like an Excel VLOOKUP operation.
+        how: {'left', 'right', 'outer', 'inner'}, default 'left'
+            How to handle the operation of the two objects.
+
+            * left: use `left` frame’s index (or column if on is specified).
+            * right: use `right`’s index.
+            * outer: form union of `left` frame’s index (or column if on is specified) with
+              right’s index, and sort it. lexicographically.
+            * inner: form intersection of `left` frame’s index (or column if on is specified)
+              with `right`’s index, preserving the order of the `left`’s one.
+        lsuffix : str, default ''
+            Suffix to use from left frame's overlapping columns.
+        rsuffix : str, default ''
+            Suffix to use from `right` frame's overlapping columns.
+
+        Returns
+        -------
+        DataFrame
+            A dataframe containing columns from both the `left` and `right`.
+
+        See Also
+        --------
+        DataFrame.merge: For column(s)-on-columns(s) operations.
+
+        Notes
+        -----
+        Parameters on, lsuffix, and rsuffix are not supported when passing a list of DataFrame
+        objects.
+
+        Examples
+        --------
+        >>> kdf1 = ks.DataFrame({'key': ['K0', 'K1', 'K2', 'K3'],
```

Comment:
Out of curiosity, why did we use a dataset different from Pandas? (https://github.com/pandas-dev/pandas/blob/v0.24.2/pandas/core/frame.py#L6694-L6815)

Reply:
In another PR, i use pandas example, but be told the example is too complex during code review.So in this PR, i simplify the redundant rows.Which way do you think is better，same pandas or simplify examlple ?